src/google-classroom-extractor/google_classroom_extractor/csv_generation/write.py

```diff
@@ -30,9 +33,16 @@ def write_csv(df_to_write: DataFrame, output_date: datetime, directory: str):
     directory: str
         is the directory the file will go in
     """
-    os.makedirs(directory, exist_ok=True)
     filename: str = output_date.strftime("%Y-%m-%d-%H-%M-%S")
-    df_to_write.to_csv(os.path.join(directory, f"{filename}.csv"), index=False)
+    path = os.path.join(directory, f"{filename}.csv")
+
+    try:
+        os.makedirs(directory, exist_ok=True)
+        df_to_write.to_csv(path, index=False)
+
+        logger.info(f"Generated file => {path}")
+    except Exception:
+        logger.exception("An exception occurred while writing file %s", path)
```

Comment:
This should really have a unit test - not covering the logging itself, but covering that the exception is caught here _and not bubbled up_. Why swallow the exception? Because if we bubble it up, then the entire application will likely stop running - whereas we want it to try running the next file in the list.